docs/book/src/developer/guide.md

```diff
@@ -221,3 +221,12 @@ Now you can [create CAPI objects][qs]!
 To test another iteration, you'll need to follow the steps to build, push, update the manifests, and apply.
 
 [qs]: https://cluster-api.sigs.k8s.io/user/quick-start.html#usage
+
+## Videos explaining CAPI architecture and code walkthrough
+
+- CAPI components and architecture
+	- https://youtu.be/npFO5Fixqcc
```

Comment:
```suggestion
	- [Cluster API Deep Dive](https://youtu.be/npFO5Fixqcc)
```

Consider formatting the links with the title of the video so it's easier to know which one is which when we add more to the list in the future

Reply:
Let's also maybe add the year and target version to these?

Reply:
@vincepri I added in the year and month of the videos, kind of hard to tell the target version from the videos though. is year/month enough? For the CAPD walkthrough I've added v1alpha4

Reply:
should we include other kubecon videos as well?

Reply:
sure, I tried adding only the latest videos, but if there's any other specific videos you want to add let me know 👍

Reply:
I could find these

https://www.youtube.com/watch?v=AeHfVFepsMg
https://www.youtube.com/watch?v=9SfuQQeeK6Q
https://www.youtube.com/watch?v=B_Zb9_Ralb4

Also, including the latest makes sense. I'll let others chime in on whether to add these or now.

Reply:
ok thanks, the second one's already in the PR, I can add in the remaining two

Reply:
@ashish-amarnath the first video is v1alpha3 version right? and what about the third?